src/ParallelAlgorithms/Amr/Actions/EvaluateRefinementCriteria.hpp

```diff
@@ -108,9 +108,21 @@ struct EvaluateRefinementCriteria {
     }
 
     const auto& policies = db::get<amr::Tags::Policies>(box);
-    amr::enforce_policies(make_not_null(&overall_decision), policies,
-                          element_id,
-                          get<::domain::Tags::Mesh<volume_dim>>(box));
+    const auto limit_types = amr::enforce_policies(
+        make_not_null(&overall_decision), policies, element_id,
+        get<::domain::Tags::Mesh<volume_dim>>(box));
+    if (policies.limits().error_beyond_limits() and
+        alg::any_of(
+            limit_types,
+            [](const std::optional<amr::Limits::LimitType>& limit_type) {
+              return limit_type.has_value();
+            })) {
+      ERROR("Tried refining beyond the AMR limits in element "
```

Comment:
Throw this error inside `amr::enforce_policies`. Then you don't need to return anything, don't need the enum, don't have to repeat the error, and it just seems easier overall.